lib/console/gateways/gateways.ex

```diff
@@ -105,4 +111,25 @@ defmodule Console.Gateways do
   def change_gateway(%Gateway{} = gateway) do
     Gateway.changeset(gateway, %{})
   end
+
+  def get_gateway_by_unique_identifier(id) do
+    case Base.decode32(id) do
+      {:ok, identifier} ->
+        case Repo.get_by(GatewayIdentifier, unique_identifier: identifier) do
+          gatewayIdentifier = %GatewayIdentifier{} -> gatewayIdentifier |> Repo.preload(:gateway)
+          nil -> :error
+        end
+      :error -> :error
+    end
+  end
+
+  defp create_gateway_identifier do
```

Comment:
given the comment below on the db table, we probably want to put this in its own context, maybe `HardwareIdentifiers` or something?